pkg/gitprovider/github.go

```diff
@@ -483,8 +486,18 @@ func (g *GitHubGitProvider) GetDefaultBranch(staticContext *StaticGitContext) (*
 
 	repo, _, err := client.Repositories.Get(context.Background(), staticContext.Owner, staticContext.Name)
 	if err != nil {
-		return nil, err
+		return nil, g.FormatError(err)
 	}
 
 	return repo.DefaultBranch, nil
 }
+
+func (g *GitHubGitProvider) FormatError(err error) error {
+	re := regexp.MustCompile(`([A-Z]+)\s(https:\/\/\S+):\s(\d{3})\s(.+)\s\[\]`)
+	match := re.FindStringSubmatch(err.Error())
+	if len(match) == 5 {
+		return fmt.Errorf("status code: %s err: %s, %s, %s", match[3], match[4], match[2], match[1])
```

Comment:
![Screenshot 2024-09-19 at 13 56 56](https://github.com/user-attachments/assets/42cc6f8c-183f-4db5-a983-01a74a42bd9b)

This is not very pretty 😄 

Please format the err message as such: `Request to {URL} failed with {message}` (no need for the http method). Apply this format to other providers as well please.